src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -392,8 +423,58 @@ int corehost_libhost_init(hostpolicy_init_t &hostpolicy_init, const pal::string_
 
 namespace
 {
-    int get_coreclr_delegate(const std::shared_ptr<coreclr_t> &coreclr, coreclr_delegate_type type, void** delegate)
+    int load_runtime_for_context(const context_handle instance)
+    {
+        if (instance == nullptr)
+            return StatusCode::InvalidArgFailure;
+
+        hostpolicy_context_t *context = static_cast<hostpolicy_context_t*>(instance);
+        std::shared_ptr<coreclr_t> coreclr;
+        return get_or_create_coreclr(context, coreclr);
+    }
+
+    int run_app_for_context(
+        const context_handle instance,
+        const int argc,
+        const pal::char_t *argv[])
     {
+        if (instance == nullptr)
+            return StatusCode::InvalidArgFailure;
+
+        hostpolicy_context_t *context = static_cast<hostpolicy_context_t*>(instance);
+        std::shared_ptr<coreclr_t> coreclr;
+        int rc = get_or_create_coreclr(context, coreclr);
+        if (rc != StatusCode::Success)
+            return rc;
+
+        return run_as_app(coreclr, *context, argc, argv);
+    }
+
+    int get_delegate_for_context(
+        const context_handle instance,
+        coreclr_delegate_type type,
+        void **delegate)
+    {
+        if (delegate == nullptr)
+            return StatusCode::InvalidArgFailure;
+
+        std::shared_ptr<coreclr_t> coreclr;
+        if (instance == nullptr)
```

Comment:
What is the scenario for the context handle to be null?
hostfxr should never do that right?

Reply:
This was trying to handle the mix of old and new APIs, so that if `hostfxr_main*` was used to load the runtime (so the active context is empty and doesn't actually have the context contract or properties), a secondary context created with the new ones could have a null context handle and use it to get a delegate.

But as we discussed, there are more issues with that, so likely as part of handling that, the context handle will never be null.